Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -0,0 +1,129 @@
+    }()
+
+    private lazy var expiryDateView: InputView = {
+        let view = InputView().disabledAutoresizingIntoConstraints()
+        view.delegate = self
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        // setup expiry DateView
+        addSubview(expiryDateView)
+        expiryDateView.setupConstraintEqualTo(view: self)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func update(style: CellTextFieldStyle?) {
+        self.style = style
+        expiryDateView.update(style: style)
+    }
+
+    private func updateErrorView(isHidden: Bool, text: String?){
+        style?.error?.isHidden = isHidden
+        style?.textfield.text = text ?? ""
+        expiryDateView.update(style: style)
+    }
+
+    func updateExpiryDate(to newDate: String?){
+        guard newDate?.count == dateFormatTextCount else {
+            updateErrorView(isHidden: false, text: newDate)
+            return
+        }
+        let subString = newDate?.split(separator: "/")
+        guard let month = subString?.first,
+              month.count == 2,
+              let monthDigit = Int(month),
+              let year = subString?.last,
+              year.count == 2,
+              let yearDigit = Int(year) else {
+            updateErrorView(isHidden: false, text: newDate)
+            return
+        }
+
+        switch cardValidator.validate(expiryMonth: monthDigit, expiryYear: yearDigit) {
+            case .success:
+                let expiryDate = ExpiryDate(month: monthDigit, year: yearDigit)
+                delegate?.update(expiryDate: expiryDate)
+                updateErrorView(isHidden: true, text: newDate)
+            case .failure(let error):
+                print(error)
+                updateErrorView(isHidden: false, text: newDate)
+        }
+    }
+}
+
+extension ExpiryDateView: TextFieldViewDelegate {
+    func textFieldShouldBeginEditing(textField: UITextField) {}
+    func textFieldShouldReturn() -> Bool {  return true }
+    func textFieldShouldEndEditing(textField: UITextField, replacementString: String) -> Bool { return true }
+
+    func textFieldShouldChangeCharactersIn(textField: UITextField, replacementString string: String) {
+        expiryDateView.textFieldContainer?.layer.borderColor = style?.textfield.focusBorderColor.cgColor
+    }
+
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+
+        /// Expiry date text format is "MM/yy"
+        /// 5 is the expected text count, for example "11/35".
+        guard range.location < dateFormatTextCount else {
+            return false
+        }
+
+        // Hide error view on remove 
+        if range.location == 4 {
+            updateErrorView(isHidden: true, text: textField.text)
+        }
+
+        if range.length > 0 {
+            if range.location == 3 {
```

Comment:
The number 3 can be added inside a constant to be more readable what is the meaning of it.